server/http_hooks.go

```diff
@@ -45,3 +51,66 @@ func (p *Plugin) handleStatus(w http.ResponseWriter, r *http.Request) {
 func (p *Plugin) handleHello(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte("Hello World!"))
 }
+
+func (p *Plugin) handleEphemeralUpdate(w http.ResponseWriter, r *http.Request) {
+	request := model.PostActionIntegrationRequestFromJson(r.Body)
+	URL := fmt.Sprintf("%s", *p.API.GetConfig().ServiceSettings.SiteURL)
+
+	count := int(request.Context["count"].(float64)) + 1
+
+	post := &model.Post{
+		Id:        request.PostId,
+		ChannelId: request.ChannelId,
+		UserId:    request.UserId,
+		Message:   fmt.Sprintf("updated ephemeral action %d", count),
+		Props: model.StringInterface{
+			"attachments": []*model.SlackAttachment{
+				{
+					Actions: []*model.PostAction{
+						{
+							Id: model.NewId(),
+							Integration: &model.PostActionIntegration{
+								Context: model.StringInterface{
+									"count": count,
+								},
+								URL: fmt.Sprintf("%s/plugins/%s/ephemeral/update", URL, manifest.Id),
+							},
+							Type: model.POST_ACTION_TYPE_BUTTON,
+							Name: "Update",
+						},
+						{
+							Id: model.NewId(),
+							Integration: &model.PostActionIntegration{
+								Context: model.StringInterface{},
+								URL:     fmt.Sprintf("%s/plugins/%s/ephemeral/delete", URL, manifest.Id),
+							},
+							Type: model.POST_ACTION_TYPE_BUTTON,
+							Name: "Delete",
+						},
+					},
+				},
+			},
+		},
+	}
+	p.API.UpdateEphemeralPost(request.UserId, post)
+
+	resp := &model.PostActionIntegrationResponse{}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write(resp.ToJson())
+
```

Comment:
Nit: Remove this line.

Reply:
the header?  removed on both that and in handleEphemeralDelete.